### PR TITLE
Make README docs use StateFlow

### PR DESCRIPTION
**Pull request by kevincianfarini**

### CODE REVIEW

**Comment by kevincianfarini** [comment]
@JakeWharton verifying this works in a real test, one second.

**Comment by JakeWharton** [comment]
You could commit the test

**Comment by kevincianfarini** [comment]
That's a good idea. I will do that first thing tomorrow. I need to sleep.

**Comment by JakeWharton** [comment]
I need to take out the trash. Then sleep.

**Comment by kevincianfarini** [comment]
Do you think this test should go in a samples directory?

**Comment by kevincianfarini** [comment]
After giving #33 another look, I think using shared flow in the docs is fine. While writing a test case I was also further convinced of this because `StateFlow` _always_ has a replay cache of 1. I think trying to work around that for the example in the documentation might be misleading. 

For example, this is what I came up with for the test case that fails with a timeout. 

```kotlin
  @Test fun timeoutHotFlowDroppedEvents() = suspendTest {
    val flow = MutableStateFlow(0)
    flow.value = 1
    assertThrows<TimeoutCancellationException> {
      flow.test {
        expectItem() // discard initial value
        assertEquals(1, expectItem())
      }
    }
  }
```

Because we're having to deal with the replay cache of size 1, we have to discard an item. So not only are we not being straightforward in the docs without users because of that, this test is also just wrong. 0 is never emitted because it's been dropped since we did `flow.value = 1`. The initial value in the `test` block _is_ 1. 

tl;dr -- The current docs are fine as of the last PR. Testing a state flow with turbine is the same as testing a shared flow with replay size of 1.

**Comment by kevincianfarini** [comment]
I'm going to close this for now. If you'd like me to commit a working test sample with shared flow I can do that. Apologies for the confusion.

**Comment by JakeWharton** [comment]
I'm satisfied with what we have. No test strictly needed.
